### PR TITLE
Feature/201/incorrect time stamp using micro on windows

### PR DESCRIPTION
**Pull request by alfonso-soto**
This pull request is coming from issue #201 . When the pull request is merged into "master" the issue will be [resolved, closed, fixed] #201

### CODE REVIEW

File: srcCpp/Infrastructure_micro.h [hunk 1]
```diff
@@ -54,6 +60,10 @@ class PerftestClock {
     RTI_INT32 clockSec;
     RTI_UINT32 clockUsec;
 
+#if defined(RTI_WIN32)
+    double PCFreq = 0.0;
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.h]: Could you add a more descriptive name? Also, add some documentation here and in the .cxx explaining that for Micro on Windows the resolution of the clock is not good enough and we will just use the native api.

**Comment by javmorcas** [comment]
I discussed this issue with Francis and we checked that indeed, in micro we use a less precise way to measure.

Add the documentation about this, and we will commit this code.

Also, add the fix for the incorrect casting.

File: srcCpp/Infrastructure_micro.h (hunk 1 above)
Comment (alfonso-soto) [reply srcCpp/Infrastructure_micro.h]: Done

File: srcCpp/Infrastructure_micro.cxx [hunk 2]
```diff
@@ -51,7 +66,22 @@ unsigned long long PerftestClock::getTimeUsec()
             &clockSec,
             &clockUsec,
             (struct OSAPI_NtpTime*)&clockTimeAux);
-    return clockUsec + 1000000 * clockSec;
+    return clockUsec + (unsigned long long)1000000 * clockSec;
+
+#else
+    /*
+        The currect clock used by Micro on Windows is measured in milliseconds,
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: comments:
```
/*
 * Line 1
 * Line 2
 */
```

File: srcCpp/Infrastructure_micro.cxx [hunk 3]
```diff
@@ -28,6 +28,19 @@ PerftestClock::PerftestClock()
     OSAPI_NtpTime_from_millisec(&clockTimeAux, 0, 0);
     clockSec = 0;
     clockUsec = 0;
+
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: You should not need the other implementation, right? No need to declare variables you are not using.

Comment (javmorcas) [reply srcCpp/Infrastructure_micro.cxx]: Or initialize

File: srcCpp/Infrastructure_micro.cxx [hunk 4]
```diff
@@ -28,6 +28,19 @@ PerftestClock::PerftestClock()
...
+#if defined(RTI_WIN32)
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: use #ifdef or #ifndef syntax

File: srcCpp/Infrastructure_micro.cxx (hunk 2 above)
Comment (javmorcas) [reply srcCpp/Infrastructure_micro.cxx]: - Ask Francis is we should create a jira issue for this lack of precision on Windows, and if he agrees, add it here.
- Reference the Microsoft page where this way to measure time is documented.

File: srcCpp/Infrastructure_micro.h [hunk 5]
```diff
@@ -54,6 +60,10 @@ class PerftestClock {
     RTI_INT32 clockSec;
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.h]: Are you using these variables if you are on Windows?

File: srcDoc/release_notes.rst [hunk 6]
```diff
@@ -334,6 +334,18 @@ This makes the function OS independent.
 At the same time, the code has been improved avoid overflowing the time for the sleeping
 period.
 
+Incorrect time stamp using micro on windows (#202)
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: I don't think we need a public release note for this, as micro was never launched officially, so you can skip this.

Comment (javmorcas) [reply srcDoc/release_notes.rst]: Remove it.

File: srcCpp/Infrastructure_micro.cxx [hunk 7]
```diff
@@ -51,7 +66,30 @@ unsigned long long PerftestClock::getTimeUsec()
             &clockSec,
             &clockUsec,
             (struct OSAPI_NtpTime*)&clockTimeAux);
-    return clockUsec + 1000000 * clockSec;
+    return clockUsec + (unsigned long long)1000000 * clockSec;
+
+#else
+    /*
+     * Micro take the timestamp by GetSystemTimeAsFileTime, this function should
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: RTI Connext DDS Micro takes*

File: srcCpp/Infrastructure_micro.cxx [hunk 8]
```diff
@@ -51,7 +66,30 @@ unsigned long long PerftestClock::getTimeUsec()
...
+     * have a resolution of 100 nanoseconds but GetSystemTimeAsFileTime is a
+     * non-realtime time & busy loop (every fast) in implementation. The system
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: every fast?

File: srcCpp/Infrastructure_micro.cxx [hunk 9]
```diff
@@ -51,7 +66,30 @@ unsigned long long PerftestClock::getTimeUsec()
...
+     * time is obtained from SharedUserData, which is fill by system on every
+     * hardware clock interruption. MICRO-2099
+     *
+     * More info here:
+     * https://docs.microsoft.com/en-us/windows/win32/sysinfo/acquiring-high-resolution-time-stamps
+     *
+     * In order to obtain enough precission for a latencyTest we are going to
+     * use the native API QueryPerformanceCounter function measured in
+     * microseconds as RTI Connext DDS Pro does.
+    */
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: space missing

File: srcCpp/Infrastructure_micro.cxx [hunk 10]
```diff
@@ -51,7 +66,30 @@ unsigned long long PerftestClock::getTimeUsec()
...
+    LARGE_INTEGER ticks;
+    QueryPerformanceCounter(&ticks);
+    return (unsigned long long)(ticks.QuadPart)/(_frequency/1000000.0);
+
+#endif
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: #ifdef and #endif indentation should be 2 spaces less than the code existent

File: srcCpp/Infrastructure_micro.cxx [hunk 11]
```diff
@@ -51,7 +66,30 @@ unsigned long long PerftestClock::getTimeUsec()
             &clockSec,
             &clockUsec,
             (struct OSAPI_NtpTime*)&clockTimeAux);
-    return clockUsec + 1000000 * clockSec;
+    return clockUsec + (unsigned long long)1000000 * clockSec;
+
+#else
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: #ifdef and #endif indentation should be 2 spaces less than the code existent

File: srcCpp/Infrastructure_micro.cxx [hunk 12]
```diff
@@ -43,6 +56,8 @@ PerftestClock &PerftestClock::getInstance()
 unsigned long long PerftestClock::getTimeUsec()
 {
 
+#ifndef RTI_WIN32
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.cxx]: #ifdef and #endif indentation should be 2 spaces less than the code existent

File: srcCpp/Infrastructure_micro.h [hunk 13]
```diff
@@ -54,6 +60,10 @@ class PerftestClock {
...
     RTI_UINT32 clockUsec;
 
+#if defined(RTI_WIN32)
+    double _frequency = 0.0;
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.h]: I think you can skip having clockSec and clockUsec:

#ifndef ...


#else 

double _fequency = 0.0;

File: srcCpp/Infrastructure_micro.cxx (hunk 7 above)
Comment (javmorcas) [reply srcCpp/Infrastructure_micro.cxx]: Not resolved

File: srcCpp/Infrastructure_micro.h [hunk 14]
```diff
@@ -19,6 +19,12 @@
   #include "PerftestSecurity.h"
 #endif
 
+#ifdef RTI_WIN32
+  #include "windows.h"
+#endif
+
+
```
Comment (javmorcas) [inline srcCpp/Infrastructure_micro.h]: Remove to new lines.

**Pull request merged**